### ISSUE TITLE
[SIP-122] Integrated monitoring and user activity dashboard

### ISSUE THREAD

**Issue by TechAuditBI** [apache/superset#27620]
### Motivation

Probably most of companies using Superset do need to gather and analyse some data about platform's health and user activity. Currently solving this task would require a bunch of work on levels from transforming raw logs to creating a dashboard. "Big" BI tools do provide such analytics out of the box so I think we should create smth like that in Superset.

### Proposed Change

The whole thing can be divided into 3 parts:

1. DB changes
Raw logs info is not suitable for building a dashboard so some sort of a datamart should be created. In case of PostgreSQL I think that it would be nice to have a materialized view with all the necessary data and a procedure to make all the transformations needed. Such a structure would provide an easy way to create an ETL to move the datamart elsewhere if needed while still being not too complicated. 

2. Dashboard itself
I'm quite sure that there is somewhat about a billion variations of this dashboard created by users so we just need to get all the popular stuff together and leave some space for specifics.

3. Deployment
We all know and love example dashboards. So maybe using similar mechanism for analytics dashboard deployment can be used. It shurely should be optional and also separate from examples. Here we need to think about many little things. Maybe there should be some parameters brought out to the config file and so on. To be discussed further.

### New or Changed Public Interfaces

Well, a new command like "superset load_monitoring" will definetely be needed. Otherwise maybe some new API endpoints would be nice but I'm not shure about this yet.

### New dependencies

Probably not needed

### Migration Plan and Compatibility

Pls help)

### Rejected Alternatives

None as for now

**Comment by sfirke** [comment]
I agree this would be a good addition to Superset, other products have this out of the box - see the respective docs from [Tableau](https://help.tableau.com/current/server/en-us/adminview_bucket.htm) and [Metabase](https://www.metabase.com/docs/latest/usage-and-performance-tools/usage-analytics) detailing what reports they offer, who can see them, etc.

I would suggest only Admin role users can view these reports, by default.  Unless that requires RBAC to be enabled - maybe someone has a concrete suggestion for how to handle the sensitivity of this reporting.

**Comment by TechAuditBI** [comment]
@supersetbot orglabel

**Comment by rusackas** [comment]
> Migration Plan and Compatibility

I don't think there's anything to migrate here, per se, but you did mention that there'll be a new command. I assume this would be run like the `load examples` command?

> Rejected alternatives

It seems like this is querying the metadata table directly... which might have performance implications depending on the size of those tables. Perhaps we should consider/reject other sorts of ETL/pipeline as an alternative process, as opposed to materialized views?

**Comment by TechAuditBI** [comment]
Yes the new command probably should run like the `load examples` one.
Data storage does require further investigation especially performance wise. Currently on our installations we use a procedure approach. Once a day during the least active period (around 3 am) the procedure runs an update script on a bunch of separate tables. This allows to avoid performance drops during peak hours. But there might be a better approach.

**Comment by TechAuditBI** [comment]
Also speaking of examples.
Maybe it would be better to split metadata and examples in separate schemas by default? Because it makes me sick looking at all the mess in a resulting db. Yes I know that there is a config parameter that allows to use a separate db connection for examples but usually it is being ignored... So maybe we should make it a bit more structural even by default.

**Comment by rusackas** [comment]
Does this need to move forward as a VOTE thread? If you have time to contribute it, I'm happy to make the vote happen.

**Comment by jbat** [comment]
re: "about a billion variations of this dashboard" ... looks like Stephan Claus and team are about to [release a new version](https://apache-superset.slack.com/archives/CK1P3L23E/p1724670077513119?thread_ts=1697053026.549729&cid=CK1P3L23E) of their dashboard, [V1 article here](https://engineering.hometogo.com/monitor-superset-usage-via-superset-c7f9fba79525).

Is there any documentation on what is logged currently in the metadb ?

And following on, "Currently solving this task would require a bunch of work on levels from transforming raw logs",  for context it would be helpful to know which log we would be referring to, at least as a starting point.

**Comment by rusackas** [comment]
We need to figure out what metadata databases we actually _support_ officially if we're going to build a built-in dashboard dependent on it.

**Comment by rusackas** [comment]
@TechAuditBI any interest in continuing to move forward with this SIP? Hopefully we can get it ready for a vote soon. This would be an amazing example to have built into Superset, but it seems there's still quite a bit of detail to flesh out in order to pull this off.

**Comment by TechAuditBI** [comment]
@rusackas We are currently designing our dash. We've chose an approach of defining the result first. I hope I'll be able to share it soon.

**Comment by rusackas** [comment]
Looking forward to more details, so we can revise the proposal accordingly and. move it forward to a VOTE.

**Comment by inviscid** [comment]
This is very much needed to track adoption and usage among stakeholders.  With some companies having multiple reporting platforms, this type of information is critical to communicate adoption over the other BI/Reporting tools.

It is also super useful when trying to understand usage patterns and the information people find most useful.

Anyway... just +1 for this when it can move forward.

**Comment by jbat** [comment]
@TechAuditBI  have you had any further developments on your monitoring dashboard?

**Comment by rusackas** [comment]
This SIP has been silent for an extended period with no mailing list discussion started. If you'd like to advance this SIP, the next step is to open a [DISCUSS] thread on `dev@superset.apache.org`. Closing due to inactivity — feel free to reopen if the mailing list discussion moves forward.
